core/src/main/java/org/infinispan/partitionhandling/impl/PartitionHandlingInterceptor.java

```diff
@@ -129,20 +132,20 @@ private boolean performPartitionCheck(InvocationContext ctx, LocalFlagAffectedCo
    }
 
    private CompletableFuture<Void> handleDataReadCommand(InvocationContext ctx, DataCommand command) {
-      Object key = command.getKey();
       return ctx.onReturn((rCtx, rCommand, rv, throwable) -> {
+         DataCommand dataCommand = (DataCommand) rCommand;
          if (throwable != null) {
-            if (throwable instanceof RpcException && performPartitionCheck(rCtx, ((DataCommand) rCommand))) {
+            if (throwable instanceof RpcException && performPartitionCheck(rCtx, dataCommand)) {
                // We must have received an AvailabilityException from one of the owners.
                // There is no way to verify the cause here, but there isn't any other way to get an invalid
                // get response.
-               throw log.degradedModeKeyUnavailable(key);
+               throw log.degradedModeKeyUnavailable(dataCommand.getKey());
```

Comment:
Hmmm, reading these comments, I see we do have a legitimate use case for a remote node throwing an exception during a clustered get. 

It also looks like throwing an exception on the first `AvailabilityException` will change our behaviour when one of the owners is `AVAILABLE` and the other is `DEGRADED`. I think that could happen if a partition is `DEGRADED` mode but one of the nodes can still send responses back to the other partition -- either because the split is still allowing some messages to go through, or because the partitions are merging but the availability mode hasn't been changed to `AVAILABLE` everywhere yet. I'm not yet sure whether I should worry about these...